src/datreant/core/trees.py

```diff
@@ -467,3 +467,17 @@ def make(self):
         self.makedirs()
 
         return self
+
+    def walk(self, topdown=True, onerror=None, followlinks=False):
+        """Generate the filenames in the tree by walking the tree either
+           top-down or bottom-up.  For each directory in the tree rooted at
+           directory top (including top itself), it yields a 3-tuple
+           (dirpath, dirnames, filenames).
+
```

Comment:
Please make this a ``numpy`` style docstring. See the surrounding docstrings for other methods for the style.